src/planscape/collaboration/models.py

```diff
@@ -0,0 +1,74 @@
+from django.contrib.gis.db import models
+from django.contrib.contenttypes.models import ContentType
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth import get_user_model
+import uuid
+
+User = get_user_model()
+
+
+class Role(models.TextChoices):
+    CREATOR = "Creator"
+    OWNER = "Owner"
+    COLLABORATOR = "Collaborator"
+    VIEWER = "Viewer"
+
+
+class Permissions(models.Model):
+    role = models.CharField(
+        choices=Role.choices,
+        max_length=16,
+        default=Role.VIEWER,
+    )
+    permission = models.CharField(max_length=60)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["role", "permission"],
+                name="unique_permission",
+            )
+        ]
+
+
+class Collaborator(models.Model):
+    uuid = models.UUIDField(default=uuid.uuid4, unique=True)
+    # the email address invited to collaborate
+    email = models.CharField(max_length=120)
+    # the user that is being added as collaborator
+    # might be empty if no user is found with the email
+    collaborator = models.ForeignKey(
+        User,
+        related_name="collaborator",
+        on_delete=models.CASCADE,
+        null=True,
+    )
+    # the role assigned
+    role = models.CharField(
+        choices=Role.choices,
+        max_length=16,
+        default=Role.VIEWER,
+    )
+    # the user that invited the collaborator
+    inviter = models.ForeignKey(
+        User,
+        related_name="inviter",
+        on_delete=models.CASCADE,
+        null=False,
```

Comment:
This might have to be nullable, if someone is the creator -- unless we want them to be their own inviter?

Reply:
That's a good point, however I was assuming we'll give `creator` role to the owner of the planning area and not through a record on this db. Do you think we should add a record here for creators?

Reply:
Good question -- I asked partly because I saw Creator listed as a Role choice, but I agree, it is probably redundant to add a record here for Creators, and redundant data always has a way of getting out of sync and becoming a problem.